### PR TITLE
Auth: Split signout_redirect_url into per provider settings

### LINKED ISSUE #73990: Split `signout_redirect_url`  into per provider settings
**Desired behavior**:

By default 

```
[auth]
signout_redirect_url = "x"
```

should be used if `auth.signout_redirect_url != ""`.

If the user belongs to an oauth provider (ex: `oauth_generic_oauth`) then on logout

```
[auth.generic_oauth]
signout_redirect_url = "x"
```

should be used if `auth.generic_oauth.signout_redirect_url != ""`, else `auth.signout_redirect_url != ""` else the default redirect.

This option should be added to all supported oauth integrations

### PR DESCRIPTION
**Pull request by venkatbvc**
**What is this feature?**

This is to support separate signout_redirect_url per oauth provider. auth.signout_redirect_url is still supported. Precedence is given to oauth provider signout url if configured both. 

**Which issue(s) does this PR fix?**:

Fixes #73990

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by chri2547 — COMMENTED** [review]
Left a docs suggestion. Thank you for the contribution.

File: docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md
Comment (venkatbvc) [reply docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md]: Addressed your suggestion

**Comment by venkatbvc** [comment]
@Jguer Can you please review and let me know your comments

File: docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md
Comment (chri2547) [reply docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md:124]: Hi @venkatbvc , I'm not sure why my entire suggestion wasn't applied. Just a few more changes. Thank you!

**Comment by Jguer** [comment]
Hey @venkatbvc , a quick code review seems to show everything is in order to get this merged. We're going to test this commit against our local sample idPs to confirm everything is working correctly and hopefully we'll just merge it :slightly_smiling_face: .

Thanks for taking the time to do this

File: pkg/api/login.go
Comment (dmihai) [inline pkg/api/login.go:257]: I'm not so sure the init of signOutRedirectUrl from this line is really needed. I think it's enough to init that var at line #261 and slightly modify the logging below to use hs.Cfg.SignoutRedirectUrl.

File: pkg/api/login.go
Comment (venkatbvc) [reply pkg/api/login.go:249]: will see and let you know.

File: pkg/api/login.go
Comment (venkatbvc) [reply pkg/api/login.go:257]: Agree, will make changes

File: pkg/api/login.go
Comment (venkatbvc) [reply pkg/api/login.go:249]: I still not able to figure how to test.

Comment (venkatbvc) [reply pkg/api/login.go:249]: @dmihai  I have added one Unit test. Let me know if it is fine

File: pkg/api/login_test.go
Comment (Jguer) [inline pkg/api/login_test.go:664]: remove comment, looks good :+1:

Comment (venkatbvc) [reply pkg/api/login_test.go]: Done.

**Comment by venkatbvc** [comment]
@Jguer  Let me know if there are any other comments.

**Comment by venkatbvc** [comment]
@Jguer Let me know if there is any pending action from myside?

File: pkg/login/social/social.go
Comment (mgyongyosi) [inline pkg/login/social/social.go:78]: Could you please change the tag to `mapstructure:"signout_redirect_url"` and move it above the `Extra` field?

Comment (venkatbvc) [reply pkg/login/social/social.go:78]: @mgyongyosi will take a look and align, incase of queries will reach out to you

Comment (venkatbvc) [reply pkg/login/social/social.go]: @mgyongyosi 
Modified as per the comment. continuous-integration/drone/pr is pending state. is it due to Milestone is set as 10.2.x?

**Comment by venkatbvc** [comment]
@mgyongyosi  Thanks for modifying the milestone. continuous-integration/drone/pr is still in pending state. It is giving below message in the details link: "**Build is blocked, please, contact repo admin in order to proceed**"
Can you please let me know what shall be done from my side?

**Comment by Jguer** [comment]
Hey @venkatbvc , I've finished testing the PR and I'm just waiting on the CI to pass before merging, no further changes should be necessary.

I confirmed it worked with generic oauth, SAML and that the legacy behavior was kept intact.

I made some small changes but I will squash the commits without co-author on merge.

Thanks for taking the time to do this, sorry that it took me some time to get back to you :slightly_smiling_face:

**Review by Jguer — APPROVED** [review]

**Review by mgyongyosi — APPROVED** [review]

**Comment by Terr4** [comment]
Hi all, is this related?
https://github.com/grafana/grafana/issues/79778

**Pull request merged**
